cpp/src/gandiva/to_date_holder.cc

```diff
@@ -83,7 +83,7 @@ int64_t ToDateHolder::operator()(ExecutionContext* context, const std::string& d
   // 1. processes date that do not match the format.
   // 2. does not process time in format +08:00 (or) id.
   int64_t seconds_since_epoch = 0;
-  if (!::arrow::internal::ParseTimestampStrptime(
+  if (!::arrow::internal::ParseTimestampStrptimeFromCString(
           data.c_str(), data.length(), pattern_.c_str(),
```

Comment:
@pitrou here data.c_str() returns null-terminated string which we pass to parsetimestamp method

Reply:
Understood. But where does that data come from? Why do you have a `std::string`?

Reply:
It is created here : https://github.com/apache/arrow/blob/295e2c11642f1b32aff1473de3f89ee02baada8a/cpp/src/gandiva/gdv_function_stubs.cc#L60

Reply:
Ah. Well, if you're concerned about performance, you shouldn't create a `std::string` there.

Reply:
Makes sense. I will update the patch to pass the data buffer directly to the orginal parsetimestamp method. The new method won't be necessary then i guess.